mypyc/irbuild/prepare.py

```diff
@@ -303,3 +306,113 @@ def prepare_non_ext_class_def(path: str, module_name: str, cdef: ClassDef,
     ):
         errors.error(
             "Non-extension classes may not inherit from extension classes", path, cdef.line)
+
+
+RegisterImplInfo = Tuple[TypeInfo, FuncDef]
```

Comment:
This can happen in a follow-up, but I think it's probably worth moving as much of the singledispatch logic as possible into its own file.